### PR TITLE
fix: fix PopupProps and improve types (close #282, #285)

### LINKED ISSUE #282: `googlePlaceId` no longer exists in version 3
Updated an app from v2 to v3 and it appears that the property `googlePlaceId` which was part of the `Options` interface as it was passed to `<Popup options={...}` no longer exists.

Seems like an oversight?

### LINKED ISSUE #285: [Bug Report] the PopupProps type are duplicated and inconsistent
The current exported `PopupProps` is defined in `src/type.ts`:

https://github.com/includable/react-native-map-link/blob/d28a2a940099481e7e3276d2de0be9331e724e51/src/type.ts#L84

However, the actual `PopupProps` in use is defined in `src/components/popup/Popup.tsx`:

https://github.com/includable/react-native-map-link/blob/d28a2a940099481e7e3276d2de0be9331e724e51/src/components/popup/Popup.tsx#L19

They are duplicated and inconsistent.

### PR DESCRIPTION
**Pull request by meteorlxy**
This PR fixes #282, fixes #285:
- Removes the duplicated `PopupProps` definition
- Improves types usage

### CODE REVIEW

File: src/components/popup/Popup.tsx [hunk 1]
```diff
@@ -37,19 +31,7 @@ interface PopupProps {
...
-    appTitles?: Record<string, string>;
-    alwaysIncludeGoogle?: boolean;
-    naverCallerName?: string;
-    latitude: number;
-    longitude: number;
-    title?: string;
-  };
-  appsWhiteList?: string[];
-  setIsVisible: (isVisible: boolean) => void;
+  options: MapLinkOptions;
```
Comment (meteorlxy) [inline src/components/popup/Popup.tsx:34]: That's the fix for #282. We should re-use `MapLinkOptions` type.

File: src/components/popup/Popup.tsx [hunk 2]
```diff
@@ -37,19 +31,7 @@ interface PopupProps {
...
-    appTitles?: Record<string, string>;
-    alwaysIncludeGoogle?: boolean;
-    naverCallerName?: string;
-    latitude: number;
-    longitude: number;
-    title?: string;
-  };
-  appsWhiteList?: string[];
-  setIsVisible: (isVisible: boolean) => void;
+  options: ShowLocationProps;
```
Comment (meteorlxy) [inline src/components/popup/Popup.tsx:34]: That's the fix for #282. We should re-use `ShowLocationProps` type.

File: docs/popup.md [hunk 3]
```diff
@@ -17,9 +17,6 @@ import { Popup } from 'react-native-map-link';
     modalProps={{ // you can put all modal props inside.
         animationIn: 'slideInUp'
     }}
-    appsWhiteList={[ /* Array of apps (apple-maps, google-maps, etc...) that you want
-    to show in the popup, if is undefined or an empty array it will show all supported apps installed on device.*/ ]}
-    appTitles: {{ /* Optional: you can override app titles. */ }}
```
Comment (meteorlxy) [inline docs/popup.md:22]: In fact `appTitles` won't take effect. It should be set in `options.appTitles`

**Review by tschoffelen — APPROVED** [review]
This is great, thanks for this addition!

**Pull request merged**